forte/processors/data_augment/algorithms/eda_ops.py

```diff
@@ -220,124 +219,143 @@
+class RandomInsertionDataAugmentOp(BaseDataAugmentationOp):
     r"""
-    Data augmentation processor for the Random Insertion operation.
+    Data augmentation operation for the Random Insertion operation.
     Find a random synonym of a random word in the sentence that is
     not a stop word. Insert that synonym into a random position in
     the sentence. Do this n times, where n = alpha * input length.
     """
 
-    def initialize(self, resources: Resources, configs: Config):
-        super().initialize(resources, configs)
-        # pylint: disable=attribute-defined-outside-init
-        self.stopwords = set(configs["stopwords"])
+    def __init__(self, configs: Union[Config, Dict[str, Any]]) -> None:
+        super().__init__(configs)
+        self.stopwords = set(self.configs["stopwords"])
+
+    def augment(self, data_pack: DataPack) -> bool:
 
-    def _augment(self, input_pack: MultiPack, aug_pack_names: List[str]):
         replacement_op = create_class_with_kwargs(
-            self.configs["data_aug_op"],
-            class_args={"configs": self.configs["data_aug_op_config"]},
+            self.configs["insertion_op"],
+            class_args={"configs": self.configs["insertion_op_config"]},
         )
-        augment_entry = get_class(self.configs["augment_entry"])
 
-        for pack_name in aug_pack_names:
-            data_pack: DataPack = input_pack.get_pack(pack_name)
-            annotations: List[Annotation] = []
-            pos = [0]
-            annos: Iterable[Annotation] = data_pack.get(augment_entry)
+        annotations: List[Annotation] = []
+        pos = [0]
+        try:
+            annos: Iterable[Annotation] = data_pack.get(
+                self.configs["augment_entry"]
+            )
             for anno in annos:
                 if anno.text not in self.stopwords:
                     annotations.append(anno)
                     pos.append(anno.end)
             if len(annotations) > 0:
                 for _ in range(ceil(self.configs["alpha"] * len(annotations))):
                     src_anno = random.choice(annotations)
-                    _, replaced_text = replacement_op.replace(src_anno)
+                    _, replaced_text = replacement_op.single_annotation_augment(
+                        src_anno
+                    )
                     insert_pos = random.choice(pos)
                     if insert_pos > 0:
                         replaced_text = " " + replaced_text
                     else:
                         replaced_text = replaced_text + " "
-                    self._insert(replaced_text, data_pack, insert_pos)
+                    self.insert_annotated_span(
+                        replaced_text,
+                        data_pack,
+                        insert_pos,
+                        self.configs["augment_entry"],
+                    )
+            return True
+        except ValueError:
```

Comment:
same as above, `try` block to large. See the Python recommendation here: https://legacy.python.org/dev/peps/pep-0008/#programming-recommendations

`Additionally, for all try/except clauses, limit the try clause to the absolute minimum amount of code necessary. Again, this avoids masking bugs.`